### DISCUSSION TITLE
Rich Text Editor

### DISCUSSION THREAD

**Discussion by as-srandall** [microsoft/fluentui-blazor#359] [Q&A]
I'm in the process of building a large enterprise-scale app on the Blazor tech stack.
I'm reviewing potential UI component libraries to see which one meets my requirements best.
This library is definitely a potential candidate, but for me the one key component missing is a **Rich Text Editor**.
I know there are plenty of rich text editors available but all seem to have their issues, limitations, constraints, and I have been unable to find exactly what I am looking for that works well in Blazor. Ideally I'm looking for an editor that can:

- easily manage **images**, pasted or selected, plus simple in-editor resizing to reduce size (like that offered by summernote).
- support for **emoji's** either in-text as I type, or selectable from a collection. I'm currently using Microsoft's fluentui-emoji collection which is perfect for my needs. I like the way the editor in Microsoft Teams manages emojis.
- support for **@mentions**. A way to perform (async) look-ups from what could be a long list of potential people.
- **autosave** functionality to either local storage or event-driven so I can update content on the backend on each event.
- easily customizable and extensible to add additional features and incorporate more complex use-cases.

I know Microsoft have many rich text editors spread over their product portfolio and not having one in the fluentui-blazor library seems like an omission. Are there any plans to add a rich text editor, or can you point me in the direction of what is felt to be the best one to investigate for inclusion in my app?

Many thanks.
Steve

**Comment by vnbaaij** [comment]
Hi Steve,

As far as I know, there a no plans for Microsoft to release a Rich Text Editor for Blazor. Due to the level of complexity involved, I don't expect one to be developed specifically or Fluent UI anytime soon.

As you said, there are a lot of closed and open-source components available already. I lack the experience with those to advise you on which one would be best to customize. Sorry that I can't be of more help here.
